python/tests/test_graphql.py

```diff
@@ -212,413 +303,1747 @@ def test_upload_graph_to_server():
+    g.add_node(5, "hamza", {"dept": "director"})
+    g.add_node(6, "haaroon", {"dept": "operations"})
+
+    work_dir = tempfile.mkdtemp()
+
+    g.save_to_file(os.path.join(work_dir, "g1"))
+
+    server = RaphtoryServer(work_dir).start()
+    client = server.get_client()
+
+    query = """mutation {
+      createGraph(
+        parentGraphName: "g1",
+        newGraphNamespace: "shivam",
+        newGraphName: "g3",
+        props: "{ \\"target\\": 6 : }",
+        isArchive: 1,
+        graphNodes: ["ben", "hamza"]
+      )
+    }"""
+    client.query(query)
+
+    query = """{
+        graph(name: "g3", namespace: "shivam") { 
+            nodes {list { 
+                name
+                properties { temporal { get(key: "dept") { values } } } 
+            }}
+            edges { list {
+                properties { temporal { get(key: "prop1") { values } } }
+            }}
+            properties { constant {
+                creationTime: get(key: "creationTime") { value }
+                lastUpdated: get(key: "lastUpdated") { value }
+                lastOpened: get(key: "lastOpened") { value }
+                uiProps: get(key: "uiProps") { value }
+                isArchive: get(key: "isArchive") { value }
+            }}
+        }
+    }"""
+
+    result = client.query(query)
+    assert result['graph']['nodes']['list'] == [
+        {'name': 'ben', 'properties': {'temporal': {'get': {'values': ['engineering']}}}},
+        {'name': 'hamza', 'properties': {'temporal': {'get': {'values': ['director']}}}}
+    ]
+    assert result['graph']['edges']['list'] == [{'properties': {'temporal': {'get': {'values': ['1']}}}}]
+    assert result['graph']['properties']['constant']['creationTime']['value'] is not None
+    assert result['graph']['properties']['constant']['lastOpened']['value'] is not None
+    assert result['graph']['properties']['constant']['lastUpdated']['value'] is not None
+    assert result['graph']['properties']['constant']['uiProps']['value'] == '{ "target": 6 : }'
+    assert result['graph']['properties']['constant']['isArchive']['value'] == 1
+
+    server.stop()
+
+
+# Update Graph with new graph name tests (save as new graph name)
+def test_update_graph_with_new_graph_name_fails_if_parent_graph_not_found():
+    work_dir = tempfile.mkdtemp()
+    server = RaphtoryServer(work_dir).start()
+    client = server.get_client()
+    query = """mutation {
+        updateGraph(
+          parentGraphName: "g0",
+          graphName: "g2",
+          graphNamespace: "shivam",
+          newGraphName: "g3",
+          props: "{{ \\"target\\": 6 : }}",
+          isArchive: 0,
+          graphNodes: ["ben"]
+        )
+    }"""
+
+    try:
+        client.query(query)
+    except Exception as e:
+        assert "Graph not found g0" in str(e), f"Unexpected exception message: {e}"
+
+    server.stop()
+
+
+def test_update_graph_with_new_graph_name_fails_if_current_graph_not_found():
```

Comment:
I realise these functions are purely for us (and not to change as part of this work) but we need to have a rethink of updateGraph as well I think